src/bootstrap/lib.rs

```diff
@@ -65,9 +65,15 @@
 //! also check out the `src/bootstrap/README.md` file for more information.
 
 #![deny(warnings)]
+#![feature(associated_consts)]
+#![feature(core_intrinsics)]
```

Comment:
Yeah I've been playing a bit with this and it is a bit tricky.

Unfortunately I'm not sure a hashmap per T will work - you still need a way to pick which hashmap T you want to look at, most easily done with a `TypeId`. Except, as discussed, `TypeId` won't work.